build-system/compile/internal-version.js

```diff
@@ -15,12 +15,22 @@
  */
 'use strict';
 
-const argv = require('minimist')(process.argv.slice(2));
+const minimist = require('minimist');
 const {gitCommitFormattedTime} = require('../common/git');
 
+// Allow leading zeros in --version_override, e.g. 0000000000001
+const argv = minimist(process.argv.slice(2), {
+  string: ['version_override'],
+});
+
 function getVersion() {
   if (argv.version_override) {
-    return String(argv.version_override);
+    const version = String(argv.version_override);
+    // #27579: What are allowed version strings...?
+    if (!/^\d{13}$/.test(version)) {
```

Comment:
Nit: Pull the version regex out into a constant for easier updating

Reply:
Agree here, but lets get this in and follow up with the change.